vmm/src/api/openapi/cloud-hypervisor.yaml

```diff
@@ -499,21 +499,28 @@ components:
           type: string
       description: Information about a PCI device
 
+    PayloadConfig:
+      type: object
+      properties:
+        kernel:
+          type: string
+        cmdline:
+          type: string
+        initramfs:
+          type: string
+      description: Payloads to boot in guest
+
     VmConfig:
       required:
-        - kernel
+        - payload
       type: object
       properties:
         cpus:
           $ref: "#/components/schemas/CpusConfig"
         memory:
           $ref: "#/components/schemas/MemoryConfig"
-        kernel:
-          $ref: "#/components/schemas/KernelConfig"
-        initramfs:
-          $ref: "#/components/schemas/InitramfsConfig"
-        cmdline:
-          $ref: "#/components/schemas/CmdLineConfig"
```

Comment:
Looks like our OpenApi is dropping the deprecated interfaces immediately while we keep them for now with the cmdline interface. Just want to make sure this is intentional.

To clarify, this is fine with Kata, as we manually bump CH version and we can make adjustments for the OpenAPI changes at the same time.

Reply:
@likebreath Great question! This was indeed intentional. This API document should document the current API. The old API still works. I know for Kata that it autogenerates from a copy of this file and that the bump to a new version of CH would also update regenerate (and so the new API can be used from the newly generated code.)

If I kept it around I think it would make the generated bindings confusing and also not make it obvious that a change is necessary.

Reply:
So how does Kata know to use the new payload API without changing their side?

Reply:
Kata has a cached copy of this file and also checks in autogenerated code. Since the old API continues to function they don't need to do anything. When they update to the next release they will pull in the new OpenAPI file, regenerate their code and find that it doesn't compile and update to reflect that.

Reply:
What I don't fully grasp yet is, Kata, when using the old cached copy, is going to make API calls with `kernel`, while Cloud Hypervisor seems to have removed the code to handle that API call. Am I missing something?

Reply:
The struct members are still their in `VmConfig` and in `VmConfig::validate()` it gets mapped to the new `PayloadConfig`. Cloud Hypervisor itself does nothing with this YAML file.

Reply:
I see. I was just checking the code more carefully before I saw your reply.

Yes, those fields are directly deserialized from the HTTP request, so CH should work just fine.